app/frontend/wallet/helpers/CachedDeriveXpubFactory.ts

```diff
@@ -1,10 +1,15 @@
 import indexIsHardened from './indexIsHardened'
+import {HARDENED_THRESHOLD, MAX_BULK_EXPORT_AMOUNT} from './../constants'
 import {derivePublic as deriveChildXpub} from 'cardano-crypto.js'
 
-function CachedDeriveXpubFactory(derivationScheme, deriveXpubHardenedFn) {
-  const derivedXpubs = {}
+const BYRON_V2_PATH = [HARDENED_THRESHOLD + 44, HARDENED_THRESHOLD + 1815, HARDENED_THRESHOLD]
 
-  async function deriveXpub(absDerivationPath) {
+type BIP32Path = number[]
+
+function CachedDeriveXpubFactory(derivationScheme, shouldExportPubKeyBulk, deriveXpubFn) {
```

Comment:
I would find it more intuitive to have the deriveXpubHardenedFn passed from above (i.e. as done originally), but the function's API would change to take potentially a list of (hardened) paths instead of a single path. Then it would be up to this function to assemble the right bulk of paths to derive and pass it down to that function. If some of the paths derived within that bulk would fail, the whole promise returned by deriveXpubHardenedFn should just fail, it's not worth dealing with partial results IMO

Reply:
My bad, the function should be called `deriveXpubsFn`. I like the idea that the logic for creating the bulk wouldn't be inside the `deriveXpubHardenedFn`.